test/philomena_web/controllers/reactivation_controller_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule PhilomenaWeb.ReactivationControllerTest do
+  use PhilomenaWeb.ConnCase, async: true
+  import Philomena.TestUtilities
+
+  alias Swoosh.Adapters.Local.Storage.Memory
+  alias Philomena.Users
+
+  setup :register_and_log_in_user
+
+  @host PhilomenaWeb.Endpoint.config(:url)[:host]
+  @port PhilomenaWeb.Endpoint.config(:http)[:port]
+
+  describe "GET /reactivations/:id" do
+    test "renders the reactivate account page", %{conn: conn} do
+      conn = delete(conn, ~p"/deactivations")
+      conn = get(conn, ~p"/reactivations/pinkie-pie-is-best-pony")
+      response = html_response(conn, 200)
+      assert response =~ "<h1>Reactivate Your Account</h1>"
+    end
+  end
+
+  describe "POST /reactivations/" do
+    test "reactivate account page works", %{conn: conn, user: user} do
+      conn = delete(conn, ~p"/deactivations")
```

Comment:
Start with a deactivated user account from a fixture instead of calling the controller here to deactivate it

See e.g. UnlockControllerTest that does:
https://github.com/philomena-dev/philomena/blob/2ee51590e68b5db8404c43ea01c55fa5083660d7/test/philomena_web/controllers/unlock_controller_test.exs#L7-L11

which itself references this fixture function:
https://github.com/philomena-dev/philomena/blob/2ee51590e68b5db8404c43ea01c55fa5083660d7/test/support/fixtures/users_fixtures.ex#L28-L32

Reply:
If I do this I will not have a token for reactivating the user in the database no? 

If I were to refactor this with the same sentiment however, I should instead start with a user and just directly call the methods to invoke the user's deactivation and reactivation rather than the current integration style testing I'm doing.

Reply:
Or should I create another user fixture that combines both approaches?

Reply:
You'll call `deliver_user_reactivation_instructions` whenever you want to get a token (see comment below, copy from UnlockControllerTest)

Reply:
Huh, I guess it doesn't actually care if the user is already in a deactivated state. Will do.

Reply:
Resolved.